Brokerages/Kraken/KrakenApi.cs

```diff
@@ -0,0 +1,707 @@
+﻿/*
+Copyright(c) 2016 Markus Trenkwalder
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using NodaTime;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+using QuantConnect.Packets;
+using QuantConnect.Securities;
+using QuantConnect.Util;
+using QuantConnect.Orders;
+using RestSharp;
+using HistoryRequest = QuantConnect.Data.HistoryRequest;
+using Order = QuantConnect.Orders.Order;
+
+namespace QuantConnect.Brokerages.Kraken
+{
+    using DataType;
+
+    public class KrakenApi : Brokerage, IDataQueueHandler
```

Comment:
Hi. First, thanks for thorough review and answers!

The `KrakenRestApi` is the "file written in another project" you had in mind. The only modification I have done on `KrakenRestApi` is renaming, second thing was taking out classes and putting them in separate files - otherwise it would be one gigantic file with all classes bundled together.

But I agree, keeping files intact might be better idea, tho 

`KrakenApi` is code I wrote. It works as a wrapper/translator for `KrakenRestApi`.

Also, KrakenApi probably doesn't need Brokerage inheritance. I am not sure - I think I just followed other exchanges implementations that seem to have layered structure.

Will check that one out and remove Brokerage / IDataQueueHandler inheritances if found out to be reduntant.